src/prompt/variable.rs

```diff
@@ -131,106 +133,70 @@ fn quoted_str(input: &str) -> IResult<&str, String, ParsingError> {
-                                        if digit > 0x7F || digit == 0 {
-                                            return Err(
-                                                format!(
-                                                    "invalid string literal: \
-                                                     invalid escape sequence '\\x{:x}' \
-                                                     (only non-null ascii allowed)", digit)
-                                            );
-                                        }
-
-                                        Ok(digit as char)
-                                    }
-                                ),
-                                value('\n', char('n')),
-                                value('\r', char('r')),
-                                value('\t', char('t')),
-                                value('\u{08}', char('b')),
-                                value('\u{0C}', char('f')),
-                                value('\\', char('\\')),
-                                value('/', char('/')),
-                                value('"', char('"')),
-                                value('\'', char('\'')),
-                            ))
-                        ),
-                        StringFragment::EscapedChar
-                    ),
-                    value(
-                        StringFragment::EscapedWS,
-                        preceded(char('\\'), nom::character::streaming::multispace1)
-                    ),
+        map_res(
+            recognize(
+                tuple((
+                    char(quote),
+                    move |str: &'a str| {
+                        let mut pos = 0;
+
+                        if str.is_empty() {
+                            return Err(Error(ParsingError::Incomplete));
+                        }
+
+                        if str.len() == 1 && str.chars().nth(0).unwrap() != quote {
+                            return Err(Error(ParsingError::Mistake {
+                                kind: None,
+                                description: "Missing end quote on string".to_string()
+                            }))
+                        }
+
+                        let chars = str.char_indices();
+
+                        let mut prev = None;
+
+                        for (i, c) in chars {
+                            if prev == None {
+                                prev = Some(c);
+                                continue;
+                            }
+
+                            if c == quote {
+                                // is it escaped?
+                                if prev.unwrap() == '\\' {
+                                    prev = Some(c);
+                                    continue;
+                                }
+
+                                // not escaped, its the end of the quoted string
+                                pos = i;
```

Comment:
I *think* that we ought to unconditionally update `pos`, so we always consume however much of the string we actually processed, even when we run off the end without a quote. The best way to do that might be to get rid of `i` and just increment `pos` at the bottom of the loop.
I'm not sure if it will actually matter, but it should allow the parser to have a better understanding of where the error happened?

Reply:
We could, but since the closure doesn't have an error case I don't think its necessary, we can just let either `char(quote)` fail or have `unquote_string` fail with its error message.

Reply:
Well, the weird thing is that `char(quote)` will fail at the "wrong position"

Reply:
Could you give an example input where it would fail in the wrong position?

Reply:
If you have `"abcdef`, `char(quote)` will fail because `a` is not a `"`, right? Which is kind of weird, because that doesn't really seem to be what the problem is here.

Reply:
![image](https://github.com/edgedb/edgedb-cli/assets/49576606/d06084d9-ec2a-4c8e-8b4c-87815bf0a9dc)

since quoted strings are never alone and are always wrapped it propigates it to the parent parser, in the case of `array<str>` it just complains about the missing `]` since the string doesn't consume anything until a quote.